embedded-graphics/src/primitives/rectangle.rs

```diff
@@ -139,6 +139,104 @@ impl Rectangle {
             None
         }
     }
+
+    /// Returns a new `Rectangle` containing the intersection of `self` and `other`.
+    ///
+    /// If no intersection is present, this method will return a zero sized rectangle with its top
+    /// left corner set to `(0, 0)`.
```

Comment:
Do we need to document that `top_left` is (0, 0)? IMO the user shouldn't care about the `top_left` corner if a rectangle is zero sized.